draft/docstrings.jl

```diff
@@ -0,0 +1,255 @@
+using Markdown
+
+struct ZZRing end
+struct ZZRingElem end
+(::ZZRing)(x) = x
+struct ZZModRing end
+struct ZZModRingElem end
+
+struct Ring end
+function polynomial_ring end
+function matrix end
+function matrix_space end
+
+@doc md"""
+    ZZRing <: Ring
+
+The ring of integers $\mathbb Z$. `ZZ = ZZRing()` is predefined.
+
+See [`ZZRingElem`](@ref) for elements and usage.
+"""
+ZZRing
+
+@doc md"""
+    ZZRingElem(x)
+
+An element of $\mathbb Z$. Also writable as [`ZZ(x)`](@ref `(::ZZRing)(x)`)
+
+# Examples
+
+```jldoctest
+julia> ZZRingElem(2)
+2
+
+julia> ZZRingElem(2)^100
+1267650600228229401496703205376
+```
+"""
+ZZRingElem
+
+@doc md"""
+    (::ZZRing)(x)
+
+An Element of $\mathbb Z$. Alias for [`ZZRingElem(x)`](@ref).
+
+# Examples
+
+```jldoctest
+julia> ZZ(2)
+2
+
+julia> ZZ(2)^100
+1267650600228229401496703205376
+```
+"""
+(::ZZRing)(x)
+
+@doc md"""
+    ZZModRing <: Ring
+
+The ring $\mathbb Z/n\mathbb Z$ for some $n$.
+
+See also [`zzModRing`](@ref) for a faster implementation in the case $n$ is small.
+
+# Examples
+
+```jldoctest
+julia> R = ResidueRing(ZZ, 42)
+Integers modulo 42
+
+julia> R(6) * R(9)
+12
+```
+"""
+ZZModRing
+
+@doc md"""
+    ZZModRingElem <: RingElem
+
+An Element of a ring $\mathbb Z/n\mathbb Z$. See [`ZZModRing`](@ref).
+"""
+ZZModRingElem
+
+# == Auto generated docstrings for matrix and polynomial spaces ==
+
+# TODO: export `Oscar.Ring`? Would be useful here, as I reference to `matrix(::Ring)`.
```

Comment:
Do you mean AbstractAlgebra.Ring?

Reply:
Correct. If we do not export `AbstractAlgebra.Ring`, I would add the `AbstractAlgebra` prefix in the docstring, which makes it clumsy.